src/components/carousel/CarouselList.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <section id="carousel">
+    <div class="container">
+      <div class="carousel-wrap" ref="carouselWrap">
+        <div class="carousel-list" ref="carouselList">
+          <carousel-item
+            v-for="(item, index) in carouselList"
+            :key="item.id"
+            :item="item"
+            :class="{ 'carousel-list__item--active': index === activeIndex }"
+            ref="carouselItem"
+          />
+        </div>
+      </div>
+      <div class="carousel__navigation">
+        <div class="carousel__pagination">
+          <span
+            v-for="(item, index) in carouselList"
+            :key="'pag-' + item.id"
+            class="pagination-bullet"
+            :class="{ 'pagination-bullet--active': index === activeIndex }"
+            ref="paginationItem"
+            @click="moveCarouselOnPaginationHandler(index)"
+          ></span>
+        </div>
+        <div class="carousel__buttons">
+          <button
+            type="button"
+            class="btn carousel-prev"
+            @click="moveCarouselHandler('back')"
+            :disabled="setAtrDisabledToBack"
+          ></button>
+          <button
+            type="button"
+            class="btn carousel-next"
+            @click="moveCarouselHandler('forward')"
+            :disabled="setAtrDisabledToFoeward"
+          ></button>
+        </div>
+      </div>
+    </div>
+  </section>
+</template>
+<script>
+import t from "vue-types";
+import CarouselItem from "./CarouselListItem.vue";
+export default {
+  props: {
+    carouselList: t.array.def([]),
+  },
+  components: {
+    CarouselItem,
+  },
+
+  data() {
+    return {
+      activeIndex: 0,
+      isResizing: false,
+      windowWidth: 0,
+    };
+  },
+  computed: {
+    setAtrDisabledToBack() {
+      return true === this.isFirstItemActive;
+    },
+    setAtrDisabledToFoeward() {
+      return true === this.isLastItemActive;
+    },
+    carouselItemsCount() {
+      return this.carouselList.length;
+    },
+    addTransition() {
+      return (this.$refs.carouselList.style.transition = this.isResizing
+        ? ""
+        : "transform 0.5s ease-in-out");
+    },
```

Comment:
This function is executing a side-effect, it can not be a computed property, consider having it as a method.  
Or even better, consider binding the style attribute the the element itself. 😉  Something like:
```js
:style="{ transition: isResizing ? 'transform 0.5s ease-in-out' : '' }"
```